src/emmetHelper.ts

```diff
@@ -521,32 +522,80 @@ export function getExpandOptions(syntax: string, emmetConfig?: object, filter?:
 		addons: addons,
 		variables: getVariables(emmetConfig['variables']),
 		snippets: customSnippetRegistry[syntax],
-		format: formatters
+		format: formatters,
+		preferences: emmetConfig['preferences']
 	};
 }
 
-function expandAbbreviationHelper(abbreviation: string, options: any) {
-	let expandedText;
-	let prefixes = ["-webkit-", "-moz-", "-ms-", "-o-"];
+function splitVendorPrefix(abbreviation: string): {prefixOptions: string, abbreviationWithoutPrefix: string} {
 	abbreviation = abbreviation || "";
-	if (isStyleSheet(options.syntax) && abbreviation[0] == '-') {
-		let tmp = expand(abbreviation.substr(1), options);
-		expandedText = tmp;
-		for (let index = 0; index < prefixes.length; index++) {
-			expandedText += "\n" + prefixes[index] + tmp;
-		}
+	if (abbreviation[0] != '-') {
+		return {
+			prefixOptions: "",
+			abbreviationWithoutPrefix: abbreviation
+		};
 	} else {
-		expandedText = expand(abbreviation, options);
+		abbreviation = abbreviation.substr(1);
+		let pref = "-";
+		if (/^[wmso]*-./.test(abbreviation)) {
+			let index = abbreviation.indexOf("-");
+			if (index > -1) {
+				pref += abbreviation.substr(0, index + 1);
+				abbreviation = abbreviation.substr(index + 1);
+			}
+		}
+		return {
+			prefixOptions: pref, 
+			abbreviationWithoutPrefix: abbreviation
+		};
+	}
+}
+
+function applyVendorPrefixes(expandedProperty: string, vendors: string, options: any) {
+	const vendorPrefixes = {'w': "webkit",'m': "moz",'s': "ms",'o': "o"};
+	expandedProperty = expandedProperty || "";
+	vendors = vendors || "";
+	if (vendors[0] == '-') {
+		if (vendors == "-") {
+			let defaultVendors = "-"
+			let preferences = options['preferences'];
+			preferences = preferences || {};
+			let property = expandedProperty.substr(0,expandedProperty.indexOf(':'));
+			property = property || expandedProperty;
+			for (const v in vendorPrefixes) {
+				let vendorProperties = preferences['css.' + vendorPrefixes[v] + 'Properties'];
+				if (vendorProperties && vendorProperties.split(', ').indexOf(property) > -1) defaultVendors += v;
```

Comment:
The assumption here is that `,` will always be followed by a space. User might not enter a space or enter multiple spaces. 
So use `vendorProperties.split(',').find(x => x.trim() == property)` instead